sql/hive/src/main/scala/org/apache/spark/sql/hive/execution/SaveAsHiveFile.scala

```diff
@@ -219,7 +219,7 @@ private[hive] trait SaveAsHiveFile extends DataWritingCommand {
     val fs: FileSystem = inputPath.getFileSystem(hadoopConf)
     var stagingPathName: String =
       if (inputPathName.indexOf(stagingDir) == -1) {
-        new Path(inputPathName, stagingDir).toString
+        new Path(inputPathName, stagingDir).toUri.getPath
```

Comment:
This drops theme:

```scala
scala> new Path("hdfs://0.0.0:8020/a/b/c").toString
res1: String = hdfs://0.0.0:8020/a/b/c

scala> new Path("hdfs://0.0.0:8020/a/b/c").toUri.getPath
res2: String = /a/b/c
```

So, if the theme is different from the default, it looks it's going to be ignored.

Reply:
+1 for @HyukjinKwon 's comment.

Reply:
Can we change this codes:
```
val inputPathUri: URI = inputPath.toUri
val inputPathName: String = inputPathUri.getPath
```
to:
`val inputPathName: String = inputPath.toString`

Reply:
Can you add a set of tests for this method `getStagingDir`?

Reply:
+1 not to use `toUri`, please see the following:
```
scala> import org.apache.hadoop.fs.Path
import org.apache.hadoop.fs.Path

scala> val f = new java.io.File("/tmp/chk %#chk")
f: java.io.File = /tmp/chk %#chk

scala> new Path(new Path(f.toURI()), "foo").toString
res0: String = file:/tmp/chk %#chk/foo

scala> new Path(new Path(f.toURI()), "foo").toUri.toString
res1: String = file:/tmp/chk%20%25%23chk/foo
```

Reply:
Yea, it's a common mistake for double-encoded problem.

Reply:
>  add a set of tests

Sorry for the late reply. 
I'll try to add some unit tests.